editor/store/effects.js

```diff
@@ -288,36 +284,30 @@ export default {
 	},
 	SETUP_EDITOR( action ) {
 		const { post, settings } = action;
-		const effects = [];
 
 		// Parse content as blocks
+		let blocks = [];
 		if ( post.content.raw ) {
-			effects.push( resetBlocks( parse( post.content.raw ) ) );
+			blocks = parse( post.content.raw );
 		} else if ( settings.template ) {
-			const blocks = map( settings.template, ( [ name, attributes ] ) => {
+			blocks = map( settings.template, ( [ name, attributes ] ) => {
 				const block = createBlock( name );
 				block.attributes = {
 					...block.attributes,
 					...attributes,
 				};
 				return block;
 			} );
-			effects.push( resetBlocks( blocks ) );
 		}
 
-		// Resetting post should occur after blocks have been reset, since it's
-		// the post reset that restarts history (used in dirty detection).
-		effects.push( resetPost( post ) );
-
 		// Include auto draft title in edits while not flagging post as dirty
+		const edits = {};
 		if ( post.status === 'auto-draft' ) {
-			effects.push( setupNewPost( {
-				title: post.title.raw,
-				status: 'draft',
-			} ) );
+			edits.title = post.title.raw;
+			edits.status = 'draft';
 		}
 
-		return effects;
+		return setupEditorState( post, blocks, edits );
```

Comment:
Maybe we can leave it for a subsequent pull request, but I think we could simplify further to merge `SETUP_EDITOR_STATE` and `SETUP_EDITOR`, including all the effect logic in the provider where it's dispatched.

Reply:
See my comment above. My idea was slightly different (I prefer yours) but yeah same direction 

> I think we can even remove setupEditorState and rely on just SETUP_EDITOR without any effect handler, the only reason I avoided this for now is to avoid calling parse in the reducer as it's not pure.